parsers/arangodb/arangodb.go

```diff
@@ -0,0 +1,196 @@
+// Package arangodb is a parser for ArangoDB logs
+package arangodb
+
+import (
+	"errors"
+	"strings"
+	"strconv"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/honeycombio/honeytail/event"
+)
+
+const (
+	iso8601UTCTimeFormat   = "2006-01-02T15:04:05Z"
+	iso8601LocalTimeFormat = "2006-01-02T15:04:05"
+
+	timestampFieldName   = "timestamp"
+	pidFieldName         = "pid"
+	logLevelFieldName    = "logLevel"
+	logTopicFieldName    = "logTopic"
+	idFieldName          = "id"
+	sourceIPFieldName    = "sourceIP"
+	methodFieldName      = "method"
+	protocolFieldName    = "protocol"
+	resCodeFieldName     = "responseCode"
+  reqBodyLenFieldName  = "reqBodyLen"
+	resBodyLenFieldName  = "resBodyLen"
+	fullURLFieldName     = "fullURL"
+	totalTimeFieldName   = "totalTime"
+)
+
+var timestampFormats = []string{
+	iso8601UTCTimeFormat,
+	iso8601LocalTimeFormat,
+}
+
+type Options struct {
+}
+
+type Parser struct {
+	conf       Options
+	lineParser LineParser
+}
+
+type LineParser interface {
+	ParseLogLine(line string) (map[string]interface{}, error)
+}
+
+type ArangoLineParser struct {
+}
+
+func firstWord(line *string) (word string, abort bool) {
+	var pos = strings.IndexByte(*line, ' ')
+	if (pos < 0) {
+		return "", true
+	}
+	word = (*line)[:pos]
+	*line = (*line)[pos+1:]
+	abort = false
+	return
+}
+
+func removeBrackets(word string) string {
+	var l int = len(word)
+  if l < 2 { return word }
+	if word[0] == '(' && word[l-1] == ')' {
+		return word[1:l-1]
+	}
+	if word[0] == '[' && word[l-1] == ']' {
+		return word[1:l-1]
+	}
+	if word[0] == '{' && word[l-1] == '}' {
+		return word[1:l-1]
+	}
+	return word
+}
+
+func removeQuotes(word string) string {
+	if len(word) == 0 { return word }
+	if word[0] == '"' {
+		word = word[1:]
+  }
+	if len(word) > 0 && word[len(word)-1] == '"' {
+		word = word[:len(word)-1]
+	}
+	return word
+}
+
+func (m *ArangoLineParser) ParseLogLine(line string) (values map[string]interface{}, err error) {
```

Comment:
Since values is only used to be an empty map on errored return, let's change it to an unnamed variable `(_ map[string]interface{}, err error)`